decode.go

```diff
@@ -578,10 +611,19 @@ func (d *Decoder) decodeStruct(dst reflect.Value, src ast.Node) error {
 						te.structFieldName = &fieldName
 					}
 					foundErr = te
-				} else {
+					continue
+				}
+
+				var ufe *unknownFieldError
+				if !xerrors.As(err, &ufe) {
 					foundErr = err
+					continue
+				}
+
+				err = d.deleteStructKeys(fieldValue, unknownFields)
```

Comment:
`deleteStructKeys`  may down performance because used reflection . 
Is this possible only when `disallowUnknownField` option is enabled?

( and if receive single `err` object , should use `if err := d.deleteStructKeys(...); err != nil {` snippet )

Reply:
I think the performance would not down, because above code checks `err` is `unknownFieldError` or not. But, it is bad readability. Do you have an idea with it?

Reply:
Hmmm...
Could you enclose it in `d.disallowUnknownField` for the time being ?
Let's do the refactor later !

Reply:
Ok! Thank you!